.circleci/config.yml

```diff
@@ -1297,41 +1320,17 @@ jobs:
       - checkout
       - attach_workspace:
           at: build
-      - run: mkdir test-cases/
-      - run: cd test-cases && ../scripts/isolate_tests.py ../test/
-      - run: cd test-cases && ../scripts/bytecodecompare/prepare_report.py ../build/solc/solc --interface standard-json --report-file ../bytecode-report-ubuntu-json.txt
-      - run: cd test-cases && ../scripts/bytecodecompare/prepare_report.py ../build/solc/solc --interface cli --report-file ../bytecode-report-ubuntu-cli.txt
-      - store_artifacts:
-          path: bytecode-report-ubuntu-json.txt
-      - store_artifacts:
-          path: bytecode-report-ubuntu-cli.txt
-      - persist_to_workspace:
-          root: .
-          paths:
-            - bytecode-report-ubuntu-json.txt
-            - bytecode-report-ubuntu-cli.txt
-      - gitter_notify_failure_unless_pr
+      - run_test_byte_code:
+          os_type: "ubuntu"
 
   b_bytecode_osx:
     <<: *base_osx
     steps:
       - checkout
       - attach_workspace:
           at: .
-      - run: mkdir test-cases/
-      - run: cd test-cases && ../scripts/isolate_tests.py ../test/
-      - run: cd test-cases && ../scripts/bytecodecompare/prepare_report.py ../build/solc/solc --interface standard-json --report-file ../bytecode-report-osx-json.txt
-      - run: cd test-cases && ../scripts/bytecodecompare/prepare_report.py ../build/solc/solc --interface cli --report-file ../bytecode-report-osx-cli.txt
-      - store_artifacts:
-          path: bytecode-report-osx-json.txt
-      - store_artifacts:
-          path: bytecode-report-osx-cli.txt
-      - persist_to_workspace:
-          root: .
-          paths:
-            - bytecode-report-osx-json.txt
-            - bytecode-report-osx-cli.txt
-      - gitter_notify_failure_unless_pr
+      - run_test_byte_code:
+          os_type: "osx"
 
   b_bytecode_win:
     <<: *base_win_cmd
```

Comment:
Now that I think of it, we could have the command handle windows and emscripten as well, using `when` blocks. This way we'd be able to share at least the part that persists and workspace and artifacts. The Python commands might actually work too given that on Windows the `/` separators are supported.

Could you try that?

Reply:
Hi @cameel yes, I think we can further share some part of the b_bytecode_win and b_bytecode_ems

currently these job are quite different versus each other,

so if we make split using small step, would that might affect the code readability?

Reply:
It would but duplication is not great either. Especially when the duplicated bits are no longer next to each other.

I mean, I'll still accept the PR even without this (especially given that you're having problems triggering the CI so it might be hard to get right) but I think I think that it would be an improvement.